### PR TITLE
Refactor tempto to use com.google.inject

### PR DESCRIPTION
**Pull request by wendigo**
This change:

- updates airbase to 140 with Guice 6.0.0
- switches injection annotation to be Guice-based
- adds minimal support for jakarta.inject annotations

### CODE REVIEW

**Comment by wendigo** [comment]
cc @electrum

File: tempto-core/pom.xml [hunk 1]
```diff
@@ -160,6 +155,19 @@
         </dependency>
 
         <!-- for testing -->
+        <dependency>
+            <groupId>javax.inject</groupId>
+            <artifactId>javax.inject</artifactId>
+            <scope>runtime</scope>
```
Comment (electrum) [inline tempto-core/pom.xml:154]: Move above with a comment
```xml
<!-- used by tests but also needed transitively -->

File: tempto-core/pom.xml [hunk 2]
```diff
@@ -160,6 +155,19 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>jakarta.inject</groupId>
+            <artifactId>jakarta.inject-api</artifactId>
+            <version>2.0.1</version>
+            <scope>runtime</scope>
```
Comment (electrum) [inline tempto-core/pom.xml:160]: This should be `test` scope since it's just for testing. Using `runtime` will make it transitive

Comment (wendigo) [reply tempto-core/pom.xml:160]: If I do that I'll get:
```
[ERROR] Found a problem with test-scoped dependency jakarta.inject:jakarta.inject-api
  Scope compile was expected by artifact: com.google.inject:guice:7.0.0
```

**Review by electrum — APPROVED** [review]

**Pull request merged**